staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -39,6 +40,583 @@ func RunTestWatch(ctx context.Context, t *testing.T, store storage.Interface) {
 	testWatch(ctx, t, store, true)
 }
 
+func basePod(podName string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+	}
+}
+
+func basePodUpdated(podName string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Status:     example.PodStatus{Phase: "Running"},
+	}
+}
+
+func basePodAssigned(podName, nodeName string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec:       example.PodSpec{NodeName: nodeName},
+	}
+}
+
+func baseNamespacedPod(podName, namespace string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: namespace},
+	}
+}
+
+func baseNamespacedPodUpdated(podName, namespace string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: namespace},
+		Status:     example.PodStatus{Phase: "Running"},
+	}
+}
+
+func baseNamespacedPodAssigned(podName, namespace, nodeName string) *example.Pod {
+	return &example.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: namespace},
+		Spec:       example.PodSpec{NodeName: nodeName},
+	}
+}
+
+// It tests watches of cluster-scoped resources.
+func TestClusterScopedWatch(ctx context.Context, t *testing.T, store storage.Interface) {
+	tests := []struct {
+		name          string
+		requestedName string // requestedName should be coexist with and equal to field selector metadata.name according to the valid input of watch request.
```

Comment:
I think I don't follow this comment - can you clarify?

Reply:
If my understanding is wrong please correct me:) When we do the watch request, we can specify the namespace either using client.Pods("kube-system"). Watch() or the field selector `metadata.namespace=kube-system`. But if we need to specify the name of a pod, we can only use the field selector like `metadata.name=pod1`. As shown in the NewRequestInfo(): https://github.com/kubernetes/apiserver/blob/a321fbfdd13785c99aeed072186df08621b838a1/pkg/endpoints/request/requestinfo.go#L233 Only when the `opts.FieldSelector != nil` and `opts.FieldSelector.RequiresExactMatch("metadata.name")`, we can have `requestInfo.Name = name`.

Reply:
Your undertanding is correct. I think just the comment needs to be clarified.

Reply:
I tried to revise the comment. If it is still not clear, can I just remove the comment?